editions/tw5.com/tiddlers/mechanisms/InfoMechanism.tid

```diff
@@ -19,6 +19,7 @@ System tiddlers in the namespace `$:/info/` are used to expose information about
 |[[$:/info/browser/screen/height]] |Screen height in pixels |
 |[[$:/info/node]] |Running under [[Node.js]]? ("yes" or "no") |
 |[[$:/info/url/full]] |<<.from-version "5.1.14">> Full URL of wiki (eg, ''<<example full>>'') |
+|[[$:/info/url/hash]] |<<.from-version "5.1.22">> Hash portion of URL of wiki (eg, ''<<example hash>>''). If empty click [[here|https://tiddlywiki.com/#InfoMechanism]]|
```

Comment:
What's the idea behind the note about the hash being empty? Why is it an external link?

Reply:
If the external link is clicked, the hash will have a value, so users can actually see something.

Reply:
OK I don't think that's clear. I don't think we need to explain anything here; this is only for users who know what the location hash is

Reply:
info.js is a startup module, which initializes all info tiddlers as "pseudo plugins", as far as I understand it. I don't see a "url change" mechanism. 

https://github.com/Jermolene/TiddlyWiki5/blob/master/core/modules/startup/info.js

I think this info shouldn't be dynamic. Dynamic changes to the views needs to come from "storyviews" and may be others

Reply:
I'm confused! All we're talking about here is the docs, and I'm suggesting that the "If empty..." part is not needed.

Reply:
But we absolutely will need a dynamic mechanism to respond to changes in the location hash in order to properly address #3967 .

Reply:
At the moment the "If empty" is needed, because we don't update the info automatically. So the hash hast to be there on page reload, otherwise the browser won't see it. 

This PR should implement a "snapshot" because that is what was requested in the [GG thread](https://groups.google.com/forum/#!topic/tiddlywiki/js4dpXZYMDg). 

I didn't want to implement something related to #3967 with this PR. 

I wanted to extend the existing mechanism because the hash element was missing.

Reply:
> At the moment the "If empty" is needed, because we don't update the info automatically. So the hash hast to be there on page reload, otherwise the browser won't see it. 

I'm saying that we shouldn't include the link that causes the location hash to be updated. It's too obscure, and I think unnecessary.

> This PR should implement a "snapshot" because that is what was requested in the [GG thread](https://groups.google.com/forum/#!topic/tiddlywiki/js4dpXZYMDg). 

I'm happy with it being a snapshot, my point was that the docs need to point this out.

> I didn't want to implement something related to #3967 with this PR. 



> I wanted to extend the existing mechanism because the hash element was missing.

Reply:
Sorry fat fingers!

> I didn't want to implement something related to #3967 with this PR. 

I'm afraid it is related whether we like it or not. It establishes an exposure of the location hash to userland, which is part of what we need for #3967.

> I wanted to extend the existing mechanism because the hash element was missing. 

It was deliberately missing. I still think we should try to go further with a proper, dynamic exposure of the location hash that allows custom actions to be taken when the hash changes.